lib/pages/home_page.dart

```diff
@@ -610,10 +578,83 @@ class _HomePageState extends State<HomePage> with TickerProviderStateMixin {
     }
   }
 
+  void _onProjectSettingsChanged() {
+    _loadSettingsFromPrefs();
+    _saveProjectSettingsToFile(_projectDir!);
+  }
+
+  void _loadSettingsFromPrefs() {
+    setState(() {
+      _robotSize = Size(
+        widget.prefs.getDouble('robotWidth') ?? 0.75,
+        widget.prefs.getDouble('robotLength') ?? 1.0,
+      );
+      _holonomicMode = widget.prefs.getBool('holonomicMode') ?? false;
+      _generateJSON = widget.prefs.getBool('generateJSON') ?? false;
+      _generateCSV = widget.prefs.getBool('generateCSV') ?? false;
+      _pplibClient = widget.prefs.getBool('pplibClient') ?? false;
+
+      if (_pplibClient) {
+        PPLibClient.initialize(widget.prefs);
+      } else {
+        PPLibClient.stopServer();
+      }
+    });
+  }
+
+  void _loadProjectSettingsFromFile(Directory projectDir) {
+    File settingsFile = File(join(projectDir.path, _settingsDir));
+
+    if (settingsFile.existsSync()) {
+      try {
+        final fileContents = settingsFile.readAsStringSync();
+        final json = jsonDecode(fileContents);
+
+        widget.prefs
+            .setDouble('robotWidth', json['robotSize']?['width'] ?? 0.75);
+        widget.prefs
+            .setDouble('robotLength', json['robotSize']?['length'] ?? 1.0);
+        widget.prefs.setBool('holonomicMode', json['holonomicMode'] ?? false);
+        widget.prefs.setBool('generateJSON', json['generateJSON'] ?? false);
+        widget.prefs.setBool('generateCSV', json['generateCSV'] ?? false);
+      } catch (error) {
+        Log.error('An error occurred while loading settings');
+      }
+    }
+
+    _loadSettingsFromPrefs();
+  }
+
+  void _saveProjectSettingsToFile(Directory projectDir) {
+    File settingsFile = File(join(projectDir.path, _settingsDir));
+
+    if (!settingsFile.existsSync()) {
+      settingsFile.createSync(recursive: true);
+    }
+
+    const JsonEncoder encoder = JsonEncoder.withIndent('  ');
+
+    Map<String, dynamic> settings = {
+      'robotSize': {
+        'width': _robotSize.width,
+        'length': _robotSize.height,
+      },
```

Comment:
I think you could just store these as `robotWidth` and `robotLength` to match the keys in shared preferences. It would also makes these two lines above cleaner:
```dart
widget.prefs.setDouble('robotWidth', json['robotSize']?['width'] ?? 0.75);
widget.prefs.setDouble('robotLength', json['robotSize']?['length'] ?? 1.0);
```